### ISSUE TITLE
`select` & `reject` don't have the same behaviour w.r.t to nested structures

### ISSUE THREAD

**Issue by YPares** [nushell/nushell#13886]
### Describe the bug

In the following:

```
~\taches> [{a: {b: b, c: c}}] | reject a.b                                                                                                                                                      21/09/24 10:06:45
╭───┬───────────╮
│ # │     a     │
├───┼───────────┤
│ 0 │ ╭───┬───╮ │
│   │ │ c │ c │ │
│   │ ╰───┴───╯ │
╰───┴───────────╯
~\taches> [{a: {b: b, c: c}}] | select a.c                                                                                                                                                      21/09/24 10:06:54
╭───┬─────╮
│ # │ a.c │
├───┼─────┤
│ 0 │ c   │
╰───┴─────╯
```
you'd expect both commands to return the same but actually `select` has a vert non-intuitive behaviour where it renames and flattens. Is this by design?

(Tested in nushell 0.98.0)

### How to reproduce

Just copy/paste the code above

### Expected behavior

I expected both commands to return the exact same table.

### Configuration

> | key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.98.0                                   |
| major              | 0                                        |
| minor              | 98                                       |
| patch              | 0                                        |
| branch             |                                          |
| commit_hash        | 6e1e824473e15eba246e1c43704c5d88fa237a17 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.79.0 (129f3b996 2024-06-10)      |
| rust_channel       | 1.79.0-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.79.0 (ffa9cf99a 2024-06-03)      |
| build_time         | 2024-09-18 08:03:04 +00:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash                   |
| installed_plugins  |                                          |

**Comment by NotTheDr01ds** [comment]
I don't think there's a separate issue on the topic, but this was discussed [here](https://github.com/nushell/nushell/issues/11218#issuecomment-2331461131) - Just interesting reading.  It's under consideration as a change in the related PR, but it's a big PR, so it could still be a while.

**Comment by YPares** [comment]
@NotTheDr01ds Yes, plus this issue is kind of a dupe of https://github.com/nushell/nushell/issues/10215 now I realize
